src/nvim/buffer_defs.h

```diff
@@ -664,12 +664,13 @@ struct file_buffer {
   char_u      *b_p_efm;         /* 'errorformat' local value */
   char_u      *b_p_ep;          /* 'equalprg' local value */
   char_u      *b_p_path;        /* 'path' local value */
-  int b_p_ar;                   /* 'autoread' local value */
+  bool b_p_ar;                  // 'autoread' local value, when present
+  bool has_b_p_ar;              // whether buffer has an 'autoread' local value
```

Comment:
You mean it is subtler than having having a variable anyone would expect be boolean (all other `int [b_]p_` are proper booleans) to actually be tristate? Or what do you think is the proper representation?
Anyway this commit (which could be broken out of this PR) is a bugfix (presumably caused by the _existing_ representation being too subtle) with added tests, which would seem to be useful for the implemenation to be changed later.